aerogear/graphback
# Add GraphQL interface support

[AEROGEAR-9814](https://issues.jboss.org/browse/AEROGEAR-9814)

## Description.

Add support for interfaces. See #191 for more information.

## Tasks

- [x] Collect interface nodes
- [x] Add interface types to generated schema
- [x] Remove query, subscription, mutation generation for interface types
- [x] Remove resolver generation for interfaces
- [x] Remove missing resolver warning in templates

## Verification Steps

Add the following to your template's schema file:

```graphql
interface Searchable {
  searchPreviewText: String!
}

interface Item {
  price: Int!
}

type Movie implements Searchable & Item {
  directory: String!
  searchPreviewText: String!
  price: Int!
}
```

Run `graphback generate && graphback db`.

Next, inspect the file `src/schema/generated.ts`. It should include the `Searchable`, `Item` interfaces. The `Movie` type should implement `Searchable` and `Item`.
